### PR TITLE
Optimize Dockerfiles for Smaller and Faster Deployments

### PR DESCRIPTION
**Pull request by dmccoystephenson**
## Changes
Some changes have been made to the Dockerfiles with the aim to enhance performance and efficiency:

1. Base Image: The base image in the Dockerfiles has been switched from Ubuntu to Alpine, resulting in smaller and faster deployments.
1. Splitting Dockerfiles: The Dockerfiles are now divided into two images - the builder image and the runtime image. The project is built within the builder container, and only the necessary files are copied to a new container for deployment.

## Testing
The changes have been thoroughly tested with the following results:

1. Successful Compilation: The project compiles and starts up successfully.
1. Passing Unit Tests: All unit tests pass within the Alpine container.
1. Dev Cluster Deployment: When deployed to the development cluster, the data gets processed as expected.

These updates are expected to contribute to a more efficient and streamlined development and deployment process.

### CODE REVIEW

**Review by drewjj — APPROVED** [review]
Looks good to me! Nice to have a slimmer image.

File: Dockerfile [hunk 1]
```diff
@@ -54,6 +51,22 @@ RUN echo "export CC=gcc" >> ~/.bashrc
...
+    bash \
+    librdkafka \
+    librdkafka-dev
+
+# copy the built files from the builder
+COPY --from=builder /asn1_codec /asn1_codec
+COPY --from=builder /build /build
+
 # Add test data. This changes frequently so keep it low in the file.
 ADD ./docker-test /asn1_codec/docker-test
```
Comment (payneBrandon) [inline Dockerfile:71]: Is this needed for a production version?

**Review by payneBrandon — COMMENTED** [review]
Just the one question

File: Dockerfile (hunk 1 above)
Comment (dmccoystephenson) [reply Dockerfile:71]: Yes I think so; I recall running into problems when I didn't copy these files.

Comment (dmccoystephenson) [reply Dockerfile]: Lines 67-68 are required. If either the `/asn1_codec` or `/build directories` are not found, the image runs into the trouble. Line 71 was not necessary, so I have removed it and the associated comment from the default dockerfile.

Comment (payneBrandon) [reply Dockerfile]: That's the one I was talking about, the "docker-test" files. Thanks Daniel!

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**